src/backend/llvm.py

```diff
@@ -162,8 +162,9 @@ def gen_comp(instr):
             if "label" in instr:
                 gen_label(instr)
             elif self.builder.block.is_terminated:
-                # Do not codegen for unreachable code
-                instr.op = "dummy"  # This doesn't really do anything
+                pass  # Do not codegen for unreachable code
```

Comment:
what is this for?

Reply:
LLVM requires basic blocks to satisfy 2 constraints:
* Must start with a label
* Must end with a terminating instruction (e.g. unconditional `br`, `ret`)

```llvm
define i32 @"main"()
{
alloca_sIcFeteBBN:
  %"v" = alloca i32
  %"tmp" = alloca i32
  br label %"entry_uvsCZgoUUv"
entry_uvsCZgoUUv:
  store i32 4, i32* %"v"
  br label %"somewhere"
  store i32 2, i32* %"v" ; This needs to be removed, as it is unreachable and doesn't belong in a basic block.
somewhere:
  %".4" = load i32, i32* %"v"
  %"tmp.1" = call i32 @"print"(i32 %".4")
  store i32 %"tmp.1", i32* %"tmp"
  %".6" = load i32, i32* %"v"
  ret i32 %".6"
}

```